### PR TITLE
SRCH-3634, SRCH-3640 added vertical nav component

### PR DESCRIPTION
**Pull request by yogesh27**
## Summary
- Added vertical navigation component
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: app/javascript/components/VerticalNav/VerticalNav.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  const [isOpen, setIsOpen] = useState([false, false]);
+  const [isOpen2, setIsOpen2] = useState([false, false]);
+
+  const onToggle = (
+    index: number,
+    setIsOpen: React.Dispatch<React.SetStateAction<boolean[]>>
+  ): void => {
+    setIsOpen((prevIsOpen) => {
+      const newIsOpen = [false, false];
+      newIsOpen[index] = !prevIsOpen[index];
```
Comment (ddzz) [inline app/javascript/components/VerticalNav/VerticalNav.tsx:16]: Right now the `index` is always `0`. Is there a reason `isOpen` has two booleans? I don't see the second boolean being used.

**Comment by ddzz** [comment]
Seeing CSS issues in the dropdowns. Also, both can be opened at the same time:
Screenshot 2023-03-29 at 8 44 53 AM

**Comment by ddzz** [comment]
Actually, I just pulled the latest commit and they no longer stay open at the same time. Still seeing the CSS issue however.

**Comment by yogesh27** [comment]
> Actually, I just pulled the latest commit and they no longer stay open at the same time. Still seeing the CSS issue however.

Hi @ddzz this can be parked for future discussion as we currently using uswds standard widths.

File: app/javascript/components/VerticalNav/VerticalNav.tsx (hunk 1 above)
Comment (yogesh27) [reply app/javascript/components/VerticalNav/VerticalNav.tsx:14]: this has been fixed.

**Review by ddzz — APPROVED** [review]
LGTM

**Pull request merged**